packages/app/routes/api/projects.js

```diff
@@ -63,41 +63,61 @@ const populateParams = [
 router.get(
   '/',
   passport.authenticate('jwt', { session: false }),
-  (req, res) => {
+  async (req, res) => {
+    log.info(
+      {
+        requestId: req.id
+      },
+      'User is getting projects'
+    )
+    let projects = await Project.find()
+    if (req.query.name) {
+      projects = projects.filter(project =>
+        project.name.toLowerCase().includes(req.query.name.toLowerCase())
+      )
+    }
+    if (req.query.projectStage) {
+      projects = projects.filter(project =>
+        project.stageOfProject
+          .toLowerCase()
+          .includes(req.query.projectStage.toLowerCase())
+      )
+    }
+    if (req.query.innovationArea) {
+      projects = projects.filter(project =>
+        project.innovationCategory
+          .toLowerCase()
+          .includes(req.query.innovationArea.toLowerCase())
+      )
+    }
+    if (req.query.thematicArea) {
+      projects = projects.filter(project =>
+        project.thematicArea
+          .toLowerCase()
+          .includes(req.query.thematicArea.toLowerCase())
+      )
+    }
+    if (req.query.sort === 'asc') {
+      projects = projects.sort((a, b) =>
+        a.name > b.name ? 1 : b.name > a.name ? -1 : 0
+      )
+    } else {
+      projects = projects.sort((a, b) =>
+        a.name < b.name ? 1 : b.name < a.name ? -1 : 0
+      )
+    }
+    projects = projects.filter(
+      (project, i) =>
+        i >= req.query.offset && i < req.query.offset + req.query.limit
+    )
     log.info(
       {
         requestId: req.id,
-        user: req.user.id
+        projects
       },
-      'Get all projects'
+      'Success getting project list'
     )
-
-    Project.find()
-      .sort({ $natural: -1 })
-      .populate(populateParams)
-      .exec((err, projects) => {
-        if (err) {
-          log.error(
-            {
-              err,
-              requestId: req.id,
-              user: req.user.id
-            },
-            'Error getting all projects'
-          )
-          return sendError(res, 503, 'Error getting projects from the database')
```

Comment:
Since you changed the Project find/filter, the error handling has been removed also. Could we make another way to handle possibles errors?